### ISSUE TITLE
Locked SaveRAM files are not handled gracefully

### ISSUE THREAD

**Issue by RetroEdit** [TASEmulators/BizHawk#1466]
I've been trying out `File > Save RAM > Flush Save Ram` and found that if the save file is locked, the command silently fails with no notice.

I've also had an exception thrown upon exiting a game through `File > Close Rom` because the SaveRAM is locked.

It would be preferable in both cases if a dialog box appeared telling the user that the file is locked for editing, and allowing them to try again or skip saving.

I'm using BizHawk Version 2.3.1 (x64) on Windows 8.1.

**Comment by zeromus** [comment]
Since I believe manual managing of saveram is the only way that it should be done and threaten to remove automatic saveram management one day without notice, the onus is on me to do a better job here with the manual management. Thanks for pointing this out

**Comment by eToThePi-i** [comment]
@zeromus I suppose by "manual", you mean *automatic* save management? I think it would still be ideal if a "Would you like to save your game?" dialogue popped up before quitting a game (or at the worst, crashing, though in those cases it might be better to create a back up and possibly warn the player about save corruption).

**Comment by zeromus** [comment]
No, I mean the opposite of automatic. That dialog will be a nuisance. It needs to appear when closing any game, not just the emulator. Everyone will turn it off. In short order it may as well not even be there. However a similar dialog is planned for the transitionary period, since nobody will be expecting this change of policy which is admittedly a huge change. 

If the emulator crashes, there's nothing we can do.

**Comment by eToThePi-i** [comment]
@zeromus I don't think I understand. Isn't that what I just said? I see your point about most people turning the dialog off, but I think it's a nice default anyway, and wasn't really commenting about that particular aspect.

And yeah, I don't expect magic with crashes per se, but there are often caught exceptions when using BizHawk.

**Comment by zeromus** [comment]
Yes that's what you said, and you called it automatic, which is the opposite of what it is.
